core/src/main/java/com/amplifyframework/storage/options/StorageUploadFileOptions.java

```diff
@@ -127,15 +105,19 @@ public static StorageUploadFileOptions defaultInstance() {
         private String contentType;
         private Map<String, String> metadata;
 
+        private Builder() {
+            this.metadata = new HashMap<>();
+        }
+
         /**
          * Configures the storage access level for the new
          * StorageUploadFileOptions instance.
          * @param accessLevel Storage access level
          * @return Current Builder instance for fluent chaining
          */
         @NonNull
-        public Builder accessLevel(@NonNull StorageAccessLevel accessLevel) {
-            this.accessLevel = Objects.requireNonNull(accessLevel);
+        public Builder accessLevel(@Nullable StorageAccessLevel accessLevel) {
```

Comment:
> So, if someone doesn't provide an accessLevel, I think our business rule is to use public, by default, right?

The agreed business rule is to have an overridable default access level in StorageCATEGORY configuration (rather than on plugin configuration level). We do not have this implemented yet, and I planned on getting to it eventually on a separate PR. In such case, I will build it into the `options.getDefaultOptions()` logic and change this back to `@NonNull`.